config/doc/ignition.yaml

```diff
@@ -380,6 +380,13 @@ root:
                       required: true
                     - name: needsNetwork
                       desc: whether or not the device requires networking.
+            - name: cex
+              desc: describes the IBM Crypto Express card configuration for the luks device.
+              children:
+                - name: enabled 
+                  desc: whether or not to use a CEX secure key to encrypt luks device.
+                - name: keyType 
+                  desc: the type of the secure key (CCA-AESDATA or CCA-AESCIPHER). Defaults to CCA-AESCIPHER.
```

Comment:
The reason for the optional key-type as below.
By default if we do not give `--key-type` it uses `CCA-AESDATA` . 
```
`CCA-AESDATA` -> support Linux upstream kernel version 4.12 or later use with  < CEX6C model card
`CCA-AESCIPHER` -> support Linux upstream kernel version 5.14 (or older version where required modules  have been backported)
```

The different secure key types have different sizes (in bytes): 

```
• CCA AES DATA: 64, 128 for XTS
• CCA AES CIPHER: 136, 272 for XTS 
• EP11 AES: 320, 640 for XTS
```

According to IBM allow an option to select key-type variant. 
1. AES CIPHER keys can be presented as default & recommended
2. AES DATA keys are less secure and should only be used if at one point in time an export of those keys will be required. AES CIPHER does not support export of keys.

In terms of Config knob `cex: true` is prettiest and easy. But I think we need to provide an option to user to choose the key-type.  

Kindly suggest.

Reply:
> But I think we need to provide an option to user to choose the key-type.

Do we? I can imagine some situations where a user could want to export the volume key (e.g. for backup or migration), but I'm not sure how desirable they are in practice vs just reprovisioning the node since Ignition is usually used in situations where that's the preferred option. Important data is usually backed up at the userspace/filesystem level instead (or e.g. mounted from dedicated storage services). ISTM like a large part of the appeal with CEX is that the keys never leave the hardware.

We could also not support it to start and see if there's strong motivation to add support for it in the future. It wouldn't be hard to add it at that point. I would still structure it as

```
cex:
  enabled: true
```

for extending it for this and anything else in the future.

Reply:
I understood, Will remove `keyType` option.

Reply:
Done